lib/plugins/aws/package/compile/events/stream.js

```diff
@@ -168,23 +168,28 @@ class AwsCompileStreamEvents {
             }
 
             const streamType = event.stream.type || EventSourceArn.split(':')[2];
-            const streamName = (function () {
-              if (EventSourceArn['Fn::GetAtt']) {
-                return EventSourceArn['Fn::GetAtt'][0];
-              } else if (EventSourceArn['Fn::ImportValue']) {
-                return EventSourceArn['Fn::ImportValue'];
-              } else if (EventSourceArn.Ref) {
-                return EventSourceArn.Ref;
-              } else if (EventSourceArn['Fn::Join']) {
+            const streamName = (function getStreamName(eventSourceArn) {
+              let result;
+              if (eventSourceArn['Fn::GetAtt']) {
+                result = eventSourceArn['Fn::GetAtt'][0];
+              } else if (eventSourceArn['Fn::ImportValue']) {
+                result = eventSourceArn['Fn::ImportValue'];
+              } else if (eventSourceArn.Ref) {
+                result = eventSourceArn.Ref;
+              } else if (eventSourceArn['Fn::Join']) {
                 // [0] is the used delimiter, [1] is the array with values
-                const name = EventSourceArn['Fn::Join'][1].slice(-1).pop();
-                if (name.split('/').length) {
-                  return name.split('/').pop();
-                }
-                return name;
+                const name = eventSourceArn['Fn::Join'][1].slice(-1).pop();
+                result = name.split('/').pop() || name;
+              } else {
+                result = eventSourceArn.split('/')[1];
               }
-              return EventSourceArn.split('/')[1];
-            })();
+
+              if (JSON.stringify(result).includes('Fn::')) {
+                return getStreamName(result);
+              }
```

Comment:
I think it'll be cleaner to do: `if (_.isObject(result)) getStreamName(result)`, and otherwise crash with meaningful error (via `ServerlessError` instance) if `result` is not a string).